storage/ibm_cos.py

```diff
@@ -0,0 +1,211 @@
+"""
+Provides the necessary interfaces to interact with IBM Cloud Object Storage.
+
+The purpose of this module is to enable RH Ceph QE team access to various development
+builds (mainly RPMs).
+
+This module depends on the availability of `cos` section in CephCI's global config file.
+The required keys in the `cos` section are
+    - endpoint
+    - api-key
+    - resource-id
+    - location-constraint
+
+Currently, the below methods are available for use
+    # upload
+    # download
+    # create_bucket
+    # set_lifecycle_rules
+"""
+import logging
+
+import ibm_boto3
+from ibm_botocore.client import ClientError, Config
+
+from utility.utils import get_cephci_config
+
+LOG = logging.getLogger(__name__)
+
+
+# Custom Exceptions
+class CreateBucketException(Exception):
+    pass
+
+
+class ApplyBucketLifeCycleException(Exception):
+    pass
+
+
+class UploadException(Exception):
+    pass
+
+
+class DownloadException(Exception):
+    pass
+
+
+class CloudObjectStorage:
+    """Interface to IBM's Cloud Object Storage."""
+
+    def __init__(self):
+        """Initialize the instance using global configuration."""
+        self._conf = get_cephci_config()["cos"]
```

Comment:
Let's document this new `cos` setting in the sample configuration file, so it's clear how users should use this.